macros/generic_tests/test_exists.sql

```diff
@@ -1,64 +1,76 @@
+{# We execute a dummy query without using any table to either return 0 or 1 records.
+For a successful test, return 0 records according to the dbt standard. #}
 {% macro test_exists(model, column_name) %}
 
 {% set table_exists = load_relation(model) is not none %}
 
-{# Test fails if 0 records are returned. #}
-{{ config(fail_calc = 'case when count(*) = 0 then 1 else 0 end') }}
+{# Set severity for the user-friendly log message when the test fails. #}
+{% if config.get('severity') == 'warn' %}
+    {% set log_category = 'UserWarning' %}
+{% elif config.get('severity') == 'error' %}
+    {% set log_category = 'UserError' %}
+{% else %}
+    {% set log_category = 'UserError' %}
+{% endif %}
 
-{# If the test is for a column and the table does not exist, we don't run the test to prevent the same error multiple times. #}
+{# If the test is for a column and the table doesn't exist, we return success to prevent the same error multiple times. #}
 {% if column_name is defined and not table_exists %}
+    {%- if target.type == 'databricks' -%}
+        select 'dummy_value' as `dummy`
+    {% else %}
+        select 'dummy_value' as "dummy"
+    {% endif %}
+    where 1 = 0
+{# If column_name is defined, the test is on a column level. #}
+{% elif column_name is defined and table_exists %}
+    {# Get columns in the relation to check if the field exists. #}
+    {%- set columns = adapter.get_columns_in_relation(model) -%}
 
-    {# Dummy query to make sure the test always returns a result #}
-    select 'dummy_value' as "dummy"
-
-{% else %}
-
-    {# Query the table info in INFORMATION_SCHEMA to check if it exists. If the test is defined on column level, also check if column exists #}
-    select *
-    from "INFORMATION_SCHEMA"."COLUMNS"
-    where "INFORMATION_SCHEMA"."COLUMNS"."TABLE_SCHEMA" = '{{ model.schema }}'
-        and "INFORMATION_SCHEMA"."COLUMNS"."TABLE_NAME" = '{{ model.name }}'
-        {%- if column_name is defined -%}
-        and "INFORMATION_SCHEMA"."COLUMNS"."COLUMN_NAME" = replace('{{ column_name }}', '"', '')
+    {%- set column_names = [] -%}
+    {%- for column in columns -%}
+        {%- if target.type == 'databricks' -%}
+            {%- set column_names = column_names.append('`' + column.name + '`') -%}
```

Comment:
We do not need to differentiate here, we are just creating a long string with the column names and don't actually execute this list on Databricks.